pyethapp/eth_service.py

```diff
@@ -168,11 +159,15 @@ def __init__(self, app):
         coinbase = app.services.accounts.coinbase
         env = Env(self.db, sce['block'])
 
-        genesis_data = sce['genesis_data'] if 'genesis_data' in sce else dict()
-        self.chain = Chain(env=env,
-                           genesis=genesis_data,
-                           coinbase=coinbase,
-                           new_head_cb=self._on_new_head)
+        genesis_data = mk_genesis_data(env)
+        # XXX: Workaround for
+        # https://github.com/ethereum/pyethereum/issues/726
+        genesis_data["extraData"] = genesis_data["extra_data"]
+        genesis_data["parentHash"] = genesis_data["prevhash"]
+        genesis_data["gasLimit"] = genesis_data["gas_limit"]
```

Comment:
Would it be ok to have it constructed here, for convenience, when it's not passed in?